pkg/streamingpromql/planning.go

```diff
@@ -497,6 +500,21 @@ func (p *QueryPlanner) nodeFromExpr(expr parser.Expr) (planning.Node, error) {
 			if err != nil {
 				return nil, err
 			}
+			matrixSelector, ok := node.(*core.MatrixSelector)
+			if ok && matrixSelector.Anchored {
+				_, supported := functions.FunctionsSupportingAnchoredMatrix[fnc]
+				if !supported {
+					// note that this error string matches a prometheus query evaluation error
```

Comment:
[nit] I'm not sure this comment is necessary - Prometheus' tests will fail if the error message is inconsistent